src/main/java/com/eprosima/fastdds/idl/templates/TypeObjectTestingTestSource.stg

```diff
@@ -0,0 +1,1101 @@
+    NameHash $flag.name$_$relation_name$_flag_name_hash = $bitmask.name$_$relation_name$_detail.name_hash();
+    MD5 $flag.name$_$relation_name$_name_hash("$flag.name$");
+    for(int i = 0;i < 4;++i)
+    {
+        ASSERT_EQ($flag.name$_$relation_name$_name_hash.digest[i], $flag.name$_$relation_name$_flag_name_hash[i]);
+    }
+    }
+    $endif$
+
+    flagId++;
+>>
+
+check_bitset_type(ctx, bitset, relation, relation_name, relation_capname) ::= <<
+    //=====check_bitset_type - $relation_name$ - $bitset.name$=====
+    $get_type_identifier(ctx=ctx, type=bitset, relation=relation, relation_name=relation_name, var_name=["type_identifier"])$
+    ASSERT_EQ(eprosima::fastrtps::types::EK_$if(relation)$COMPLETE$else$MINIMAL$endif$, $bitset.name$_$relation_name$_type_identifier->_d());
+
+    $get_type_object(ctx=ctx, type=bitset, relation=relation, relation_name=relation_name, var_name=["typeobject"])$
+    ASSERT_EQ(eprosima::fastrtps::types::EK_$if(relation)$COMPLETE$else$MINIMAL$endif$, $bitset.name$_$relation_name$_typeobject->_d());
+
+    $relation_capname$TypeObject $bitset.name$_$relation_name$_object = $bitset.name$_$relation_name$_typeobject->$relation_name$();
+    ASSERT_EQ(TK_BITSET, $bitset.name$_$relation_name$_object._d());
+
+    $relation_capname$BitsetType $bitset.name$_$relation_name$_bitset = $bitset.name$_$relation_name$_object.bitset_type();
+    //BitsetTypeFlag bitset_flags;// Unused. No flags apply
+
+    $relation_capname$BitsetHeader $bitset.name$_$relation_name$_header = $bitset.name$_$relation_name$_bitset.header();
+
+    $if(relation)$
+    CompleteTypeDetail $bitset.name$_$relation_name$_detail = $bitset.name$_$relation_name$_header.detail();
+    $check_optional_detail_annotations(ctx=ctx, object=bitset)$
+    QualifiedTypeName $bitset.name$_$relation_name$_detail_type_name = $bitset.name$_$relation_name$_detail.type_name();
+    ASSERT_EQ("$bitset.scopedname$", $bitset.name$_$relation_name$_detail_type_name);
+    $else$
+    //MinimalTypeDetail detail;// Empty. Available for future extension
+    $endif$
+
+    $relation_capname$BitfieldSeq $bitset.name$_$relation_name$_bitfieldseq = $bitset.name$_$relation_name$_bitset.field_seq();
+>>
+
+check_bitfield_type(ctx, bitfield, relation, relation_name, relation_capname) ::= <<
+    //=====check_bitfield_type - $relation_name$ - $bitfield.name$=====
+    $relation_capname$Bitfield field_$bitset.name$_$relation_name$_$bitfield.name$ = $bitset.name$_$relation_name$_bitfieldseq.at(bitfieldId);
+    CommonBitfield $bitset.name$_$relation_name$_common_$bitfield.name$_bitset_bitfield = field_$bitset.name$_$relation_name$_$bitfield.name$.common();
+    
+    unsigned short $bitset.name$_$relation_name$_common_$bitfield.name$_bitset_bitfield_position = $bitset.name$_$relation_name$_common_$bitfield.name$_bitset_bitfield.position();
+    ASSERT_EQ(bitfieldId, $bitset.name$_$relation_name$_common_$bitfield.name$_bitset_bitfield_position);
```

Comment:
This logic does not have sense. Bitfields are characterized by an initial position and by a bitcount, so the next position must be at least `position + bitcount`.